### PR TITLE
RavenDB-14862 Support for Documents Compression in the Studio

### PR DESCRIPTION
**Pull request by Danielle9897**

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/CompressionHandler.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+using System.Net;
+using System.Threading.Tasks;
+using Raven.Client.ServerWide;
+using Raven.Server.Routing;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers
+{
+    public class CompressionHandler : DatabaseRequestHandler
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/CompressionHandler.cs]:
```suggestion
    public class DocumentsCompressionHandler : DatabaseRequestHandler
```

Comment (Danielle9897) [reply src/Raven.Server/Documents/Handlers/CompressionHandler.cs]: done

**Review by arekpalinski — APPROVED** [review]

File: src/Raven.Studio/typescript/viewmodels/database/settings/documentsCompression.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+import appUrl = require("common/appUrl");
+import database = require("models/resources/database");
+import eventsCollector = require("common/eventsCollector");
+import collectionsTracker = require("common/helpers/database/collectionsTracker");
+import getDocumentsCompressionConfigurationCommand = require("commands/database/documents/getDocumentsCompressionConfigurationCommand");
+import saveDocumentsCompressionCommand = require("commands/database/documents/saveDocumentsCompressionCommand");
+
+class documentsCompression extends viewModelBase {
+
+    allExistingCollections:KnockoutComputed<Array<string>>;
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/settings/documentsCompression.ts]: space after :

File: src/Raven.Studio/typescript/viewmodels/database/settings/documentsCompression.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    canActivate(args: any): boolean | JQueryPromise<canActivateResultDto> {
+        return $.when<any>(super.canActivate(args))
+            .then(() => {
+                const deferred = $.Deferred<canActivateResultDto>();
+
+                this.fetchCompressionConfiguration(this.activeDatabase())
+                    .done(() => deferred.resolve({ can: true }))
+                    .fail(() => deferred.resolve({ redirect: appUrl.forDatabaseRecord(this.activeDatabase()) }));
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/settings/documentsCompression.ts:53]: why redirect to this view?

File: src/Raven.Studio/typescript/viewmodels/database/settings/documentsCompression.ts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+        new saveDocumentsCompressionCommand(this.activeDatabase(), dto)
+            .execute()
+            .done(() => this.dirtyFlag().reset())
+            .always(() => this.spinners.save(false));
+    }
+
+    toDto(): Raven.Client.ServerWide.DocumentsCompressionConfiguration {
+        return {
+            Collections: this.collectionsToCompress().sort(),
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/settings/documentsCompression.ts]: I don't think we need to sort that

File: src/Raven.Studio/wwwroot/App/views/database/settings/documentsCompression.html [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+                                    </a>
+                                </li>
+                            </ul>
+                        </div>
+                        <button class="btn btn-success" data-bind="click: $root.addCollection, enable: collectionToAdd, attr: { title: collectionToAdd() ? 'Click to add collection to the compressed collections list' : '' }">
+                            <i class="icon-plus"></i>
+                            <span>Add</span>
+                        </button>
+                        <div class=" margin-left margin-left-sm">
+                            <button class="btn btn-info" data-bind="click: $root.addAllCollections, enable: canAddAllCollections, attr: { title: canAddAllCollections() ? 'Add all collections in the database to the compressed collecions list' : '' }">
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/App/views/database/settings/documentsCompression.html]: I think we don't need logic for title - as this is always the truth

File: src/Raven.Studio/wwwroot/App/views/database/settings/documentsCompression.html [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+                        </div>
+                    </div>
+                </div>
+            </div>
+            <div class="panel">
+                <div class="padding padding-lg">
+                    <div class="toggle toggle-primary" title="All revisions for all collections will be compressed when toggled on.">
+                        <input id="revisionsToggle" class="styled" type="checkbox" data-bind="checked: compressRevisions">
+                        <label for="revisionsToggle">
+                            Compress Revisions
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/App/views/database/settings/documentsCompression.html]: I wonder if we should remove title and update label to contain info that we about to compress reivison from *all* collections?

File: src/Raven.Studio/wwwroot/Content/css/pages/document-compression.less [hunk 7]
```diff
@@ -0,0 +1,11 @@
+﻿#documentsCompression {
+    
+    .delete-btn {
+        margin-top: 3px;
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/Content/css/pages/document-compression.less]: do we need that?

Comment (Danielle9897) [reply src/Raven.Studio/wwwroot/Content/css/pages/document-compression.less]: we don't - I forgot to remove - done

File: src/Raven.Studio/wwwroot/App/views/database/settings/documentsCompression.html (hunk 6 above)
Comment (Danielle9897) [reply src/Raven.Studio/wwwroot/App/views/database/settings/documentsCompression.html]: using `Compress Revisions for All Collections`

File: src/Raven.Studio/wwwroot/App/views/database/settings/documentsCompression.html (hunk 5 above)
Comment (Danielle9897) [reply src/Raven.Studio/wwwroot/App/views/database/settings/documentsCompression.html]: done

File: src/Raven.Studio/typescript/viewmodels/database/settings/documentsCompression.ts (hunk 4 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/settings/documentsCompression.ts]: done

File: src/Raven.Studio/typescript/viewmodels/database/settings/documentsCompression.ts (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/settings/documentsCompression.ts]: done

File: src/Raven.Studio/typescript/viewmodels/database/settings/documentsCompression.ts (hunk 3 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/settings/documentsCompression.ts:53]: leaving this - as discussed

**Review by ml054 — APPROVED** [review]

**Pull request merged**